### PR TITLE
[FIX] : Fixes the entrypoint component bugs 🐛 

### LINKED ISSUE #199: [BUG] : Minor bugs related to the entry points component
This issue highlights minor bugs discovered by @lrasmus and myself regarding the newly introduced entry points component. These bugs include the component's inability to navigate to entrypoint files in the asset tree due to the usage of Unix-based paths. Additionally, it addresses some minor UI fixes related to the gap between asset details and the entry points accordion, as well as the visibility of the root URI when the entry points component is absent.

### PR DESCRIPTION
**Pull request by AdiAkhileshSingh15**
## Changes Overview

This pull request resolves issues with the entrypoints component by implementing several key changes:

- **Selective Rendering of File Path Separator:** Adjustments made to render the file path separator based on the operating system.
- **UI fixes:** Rendering root URI within the entrypoints component. Along with adjusting the gap between the asset details and entrypoints accordions.
- **Additional Unit Tests:** Added unit tests specifically for Windows-based asset file paths.

### Related Issue 🐛
Fixes #199

### Checklist ✅

The PR was subjected to the following checks to ensure functionality remained intact:

- [x] `yarn test` - Passed all tests without errors.
- [x] `yarn dev` - No UI failures were observed.
- [x] `yarn build` - The build process completed successfully.
- [x] `yarn dev` - No new console errors detected.

### Reviewer Checklist

The reviewer is requested to:

- [x] Test all modifications locally on a Windows system.
- [x] Assess and approve the UI changes implemented.

### Reviewer
@lrasmus

### CODE REVIEW

File: app/utils/asset.js [hunk 1]
```diff
@@ -120,16 +120,18 @@ export default class AssetUtil {
     const descendantsList = [];
     // check if the asset uri has a descendant root uri in it
     if (assetUri && rootUri && assetUri.includes(rootUri) && assetUri.indexOf(rootUri) === 0) {
+      // because for windows the path seperator is '\'
+      const sep = process.platform === 'win32' ? '\\' : '/';
```
Comment (lrasmus) [inline app/utils/asset.js:124]: I think you can use the built-in `path.sep` instead of deriving it yourself (https://nodejs.org/docs/latest/api/path.html#pathsep)

File: app/components/ProjectEntryPoint/ProjectEntryPoint.js [hunk 2]
```diff
@@ -35,10 +35,13 @@ const projectEntryPoint = (props) => {
     findEntryPoints(assets, entryPointsList);
   }
 
+  const sep = process.platform === 'win32' ? '\\' : '/';
```
Comment (lrasmus) [inline app/components/ProjectEntryPoint/ProjectEntryPoint.js:38]: I think you can use the built-in `path.sep` instead of deriving it yourself (https://nodejs.org/docs/latest/api/path.html#pathsep)

File: test/utils/asset.spec.js [hunk 3]
```diff
@@ -391,6 +391,49 @@ describe('services', () => {
       it('should return an empty array if the asset URI contains root URI but not as a prefix', () => {
         expect(AssetUtil.findAllDescendantAssetsByUri('/Test/Rt/Child', '/Rt')).toBeArrayOfSize(0);
       });
+      it.onWindows('should return an empty array if the asset URI is not specified', () => {
```
Comment (lrasmus) [inline test/utils/asset.spec.js:394]: Thank you for adding the Windows tests.  Unfortunately these are failing for me on my machine.  Did all tests pass for you?

**Review by lrasmus — CHANGES_REQUESTED** [review]
Main thing will be reviewing the Windows-specific tests, which I'm seeing some failing.

File: test/utils/asset.spec.js (hunk 3 above)
Comment (AdiAkhileshSingh15) [reply test/utils/asset.spec.js:394]: I actually tested them on unix, which now I think have been the best choice for validating Windows-specific tests. I assumed that by using the `.onWindows` or `.onMac` annotations, the tests would execute independently of the underlying system. 
But, am still scratching my head for how did they all pass, given that I passed the windows based filepaths in testcases.
Screenshot from 2024-04-03 20-34-37

File: app/components/ProjectEntryPoint/ProjectEntryPoint.js (hunk 2 above)
Comment (AdiAkhileshSingh15) [reply app/components/ProjectEntryPoint/ProjectEntryPoint.js:38]: Ohk, I guess this would surely work. Thanks

File: test/utils/asset.spec.js (hunk 3 above)
Comment (AdiAkhileshSingh15) [reply test/utils/asset.spec.js:408]: My bad. I assumed that by using the `.onWindows` or `.onMac` annotations, the tests would execute independently of the underlying system. I tested on Unix

Comment (lrasmus) [reply test/utils/asset.spec.js:408]: No worries!

**Comment by AdiAkhileshSingh15** [comment]
@lrasmus I've made some changes, please review them when you get a change, I believe it'll work fine now. The `findAllDescendantAssetsByUri` method was actually missing some edge cases, which I detected, and are now fixed. Tests should run fine as well, though I've not tested them on windows machine for now. If you can do that, it'd be great. 😄

**Review by lrasmus — APPROVED** [review]
All changes look good!
